cms/djangoapps/contentstore/signals.py

```diff
@@ -0,0 +1,19 @@
+""" receiver of course_published events in order to trigger indexing task """
+from datetime import datetime
+from pytz import UTC
+
+from django.dispatch import receiver
+
+from xmodule.modulestore.django import SignalHandler
+from contentstore.courseware_index import indexing_is_enabled
+
+
+@receiver(SignalHandler.course_published)
+def listen_for_course_publish(sender, course_key, **kwargs):  # pylint: disable=unused-argument
+    """
+    Receives signal and kicks off celery task to update search index
+    """
+    # import here, because signal is registered at startup, but items in tasks are not yet able to be loaded
+    from .tasks import update_search_index
+    if indexing_is_enabled():
+        update_search_index.delay(unicode(course_key), datetime.now(UTC))
```

Comment:
@martynjames what's the reason for converting `course_key` to unicode here, than reconstructing it from string [in the task itself](https://github.com/edx/edx-platform/pull/7313/files#diff-e5cb44a7b9f87fcd4099b29724da5d24R87). I'm asking this because I'm about adding another task and signla to index library, and it seems a bit weird to replicate this.